control_plane/src/storage_controller.rs

```diff
@@ -415,8 +501,42 @@ impl StorageController {
         Ok(())
     }
 
-    pub async fn stop(&self, immediate: bool) -> anyhow::Result<()> {
-        background_process::stop_process(immediate, COMMAND, &self.pid_file())?;
+    pub async fn stop(&self, stop_args: NeonStorageControllerStopArgs) -> anyhow::Result<()> {
+        background_process::stop_process(
+            stop_args.immediate,
+            COMMAND,
+            &self.pid_file(stop_args.instance_id),
+        )?;
+
+        let storcon_instances = self.env.get_storage_controller_instances().await?;
+        for (instance_id, instanced_dir_path) in storcon_instances {
+            if instance_id == stop_args.instance_id {
+                continue;
+            }
+
+            let pid_file = instanced_dir_path.join("storage_controller.pid");
+            let pid = tokio::fs::read_to_string(&pid_file)
+                .await
+                .map_err(|err| {
+                    anyhow::anyhow!("Failed to read storcon pid file at {pid_file:?}: {err}")
+                })?
+                .parse::<i32>()
+                .expect("pid is valid i32");
+
+            let other_proc_alive = match procfs::process::Process::new(pid) {
+                Ok(proc) => proc.is_alive(),
+                Err(procfs::ProcError::NotFound(_)) => false,
+                Err(_) => {
+                    anyhow::bail!("Failed to inspect storage controller proc {pid}")
+                }
+            };
```

Comment:
I think we have already some pidfile handling existing and there we use kill(0) to check if the pid is alive, perhaps we could use them here and not add procfs?

Reply:
Added the macos label, it will be taken into account on next push.

Reply:
Ah nice. I can use that instead of adding a new dependency.

Reply:
Done in https://github.com/neondatabase/neon/pull/8587/commits/ae5944f8b873bbc4a1aa55fdb1488001fe9448dd - thanks for the suggestion